lib/ably/realtime/channel.rb

```diff
@@ -179,15 +179,28 @@ def presence
         @presence
       end
 
-      # Return the message history of the channel
+      # Return the message history of the channel.
+      #
+      # Once attached to a channel, you can retrieve messages published on the channel before the
+      # channel was attached with the option <tt>:end => :before_attach</tt>.  This is very useful for
+      # developers who wish to display both new real-time messages as well as historical messages with
+      # the guarantee that no messages have been missed.
+      #
       #
       # @param (see Ably::Rest::Channel#history)
       # @option options (see Ably::Rest::Channel#history)
+      # @option options [Integer,Time,Symbol]   :end        Time or millisecond since epoch, or +:before_attach+ for all messages published before this channel has been attached.
       #
       # @yield [Ably::Models::PaginatedResource<Ably::Models::Message>] An Array of {Ably::Models::Message} objects that supports paging (#next_page, #first_page)
       #
       # @return [Ably::Util::SafeDeferrable]
       def history(options = {}, &callback)
+        if options[:end] == :before_attach
+          raise ArgumentError, 'option :before_attach cannot be specified if the channel is not attached' unless attached?
```

Comment:
> Raising an exception here is an inconvenience because the program can continue

The program can continue, but the history request may continue with unpredictable results for a developer.  Will it timeout if the attach does not happen, if the connection drops should this fail or wait for it to reconnect and then attach, if whilst in the `attaching` state someone tells it to detach should it fail, how long should it wait for the channel to attach etc.  There are way too many permutations that we need to think about make assumptions for, whereas if we simply raise an exception then we leave that decision to the developer, which is certainly my preference.